File: Examples/Conversations/Obj-C/ConversationsExample/AnswersViewController.m
//
//  AnswersViewController.m
//  ConversationsExample
//
//  Copyright © 2016 Bazaarvoice. All rights reserved.
//

#import "AnswersViewController.h"
#import "MyAnswerTableViewCell.h"

@interface AnswersViewController () <UITableViewDataSource>

@property(weak, nonatomic) IBOutlet BVAnswersTableView *answersTableView;

@end

@implementation AnswersViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  self.answersTableView.dataSource = self;
  self.answersTableView.estimatedRowHeight = 44;
  self.answersTableView.rowHeight = UITableViewAutomaticDimension;
  [self.answersTableView
                 registerNib:[UINib nibWithNibName:@"MyAnswerTableViewCell"
                                            bundle:nil]
      forCellReuseIdentifier:@"MyAnswerTableViewCell"];
}

#pragma mark UITableViewDatasource

- (NSString *)tableView:(UITableView *)tableView
    titleForHeaderInSection:(NSInteger)section {
  return @"Questions Responses";
}

- (NSInteger)tableView:(UITableView *)tableView
    numberOfRowsInSection:(NSInteger)section {
  return [self.question.includedAnswers count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView
         cellForRowAtIndexPath:(NSIndexPath *)indexPath {

  MyAnswerTableViewCell *cell =
      [tableView dequeueReusableCellWithIdentifier:@"MyAnswerTableViewCell"];

  cell.answer = [self.question.includedAnswers objectAtIndex:indexPath.row];

  return cell;
}

@end

File: Examples/Conversations/Swift/ConversationsExample/AnswersViewController.swift
//
//  AnswersViewController.swift
//  Conversations
//
//  Copyright © 2016 Bazaarvoice. All rights reserved.
//

import UIKit
import BVSDK

class AnswersViewController: UIViewController, UITableViewDataSource, UITableViewDelegate  {
  
  @IBOutlet weak var answersTableView: BVAnswersTableView!
  var question : BVQuestion?
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    answersTableView.dataSource = self
    answersTableView.delegate = self
    answersTableView.estimatedRowHeight = 68
    answersTableView.rowHeight = UITableViewAutomaticDimension
    answersTableView.register(UINib(nibName: "MyAnswerTableViewCell", bundle: nil), forCellReuseIdentifier: "MyAnswerTableViewCell")
    
  }
  
  // MARK: UITableViewDatasource
  
  func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
    return "Answer Responses"
  }
  
  
  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
    return question!.answers.count
  }
  
  func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
    let cell = tableView.dequeueReusableCell(withIdentifier: "MyAnswerTableViewCell")! as! MyAnswerTableViewCell
    
    cell.answer = question!.answers[indexPath.row];
    
    return cell
  }
  
}

